server/balancer.go

```diff
@@ -0,0 +1,316 @@
+package server
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/kvproto/pkg/raftpb"
+)
+
+// Operator is the interface to do some operations.
+type Operator interface {
+	// Do does the operator, if finished then return true.
+	Do(req *pdpb.RegionHeartbeatRequest) (bool, *pdpb.RegionHeartbeatResponse)
```

Comment:
using `region* metapb.Region, leader* metapb.Peer` is better for argument.

using `*pdpb.ChangePeer` is better for return.